doc/SONiC_Port_Mirroring_HLD.md

```diff
@@ -0,0 +1,319 @@
+
+To support planned system warm boot.
+To support SWSS docker warm boot.
+
+
+# 2 Functionality
+
+Refer section 1
+
+## 2.2 Functional Description
+Refer section 1.1
+
+## 2.3 Functional Description
+
+Mirroring to destination VLAN (RSPAN) is not supported in this release.
+
+# 3 Design
+## 3.1 Overview
+
+
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+
+Existing table PORT_MIRROR_TABLE is enhanced to accept new source and destination configuration options  in the configuration database. This table is filled by the management framework.
+
+#### CONFIG_PORT_MIRROR_TABLE
+
+    ;Configure SPAN/ERSPAN mirror session.
+    ;storm control type - broadcast / unknown-unicast / unknown-multicast
+    key       = PORT_MIRROR_TABLE:mirror_session_name ; mirror_session_name is
+                                                      ; unique session
+                                                      ; identifier
+    ;field  = value
+    destination_port = PORT_TABLE:ifname    ; ifname must be unique across PORT TABLE.
+    source_port = PORT_TABLE:ifname    ; ifname must be unique across PORT,INTF,LAG TABLES
+    direction     = ingress or egress or both           ; Direction ingress or egress or both.
+
+    mirror_session_name = 1*255VCHAR
+
+### 3.2.2 APP_DB
+No tables are introduced in APP_DB
+### 3.2.3 STATE_DB
+No tables are introduced in STATE_DB.·
+
+### 3.2.4 ASIC_DB
+No changes are introduced in ASIC_DB.·
+### 3.2.5 COUNTER_DB
+No changes are introduced in COUNTER_DB.·
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+
+Mirror Orchestration agent is modified to support this feature:
+   - Handle both SPAN and ERSPAN sessions separately·
+   - No changes to ERSPAN functionality.
+   - Configure mirror session based on CONFIG_DB parameters.
+   - Port mirror session is activated 
+   - Populates the mirror attribute SAI structures and pushes the entry to ASIC_DB.·
+
+## 3.4 SAI
+Mirror SAI interface APIs are already defined. More details about SAI API and attributes are described below SAI Spec @
+
+https://github.com/opencomputeproject/SAI/blob/master/inc/saimirror.h
+
+## 3.5 CLI
+### 3.5.1 Data Models
+Custom Yang model will be introduced for this feature.
+
+### 3.5.2 Configuration Commands
+
+Existing mirror session commands are enhanced to support this feature.
+
+    # Modify existing ERSPAN configuration as below.
+    config mirror_session add erspan <session-name> <src_ip> <dst_ip> <gre> <dscp>  [ttl] [queue]
+
+    #Configure Destination only span mirror session.
+    config mirror_session add span <session-name> <destination_ifName>
+
+    # Modify existing ERSPAN configuration to accept source port and direction
+    config mirror_session add erspan <session-name> <src_ip> <dst_ip> <gre> <dscp>  [ttl] [queue] [src_port] [rx/tx/both]
+
+    #Configure Port mirror span mirror session.
+    config mirror_session add span <session-name> <destination_ifName> <source_ifName> <rx/tx/both>
```

Comment:
is policer going to supported in span session?

Reply:
Yes. Policer is supported.